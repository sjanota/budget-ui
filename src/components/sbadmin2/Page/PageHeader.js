import React from 'react';
import PropTypes from 'prop-types';
import { withDictionary } from '../language';

function PageHeader({ title }) {
  return <h1 className="h3 mb-4 text-gray-800">{title}</h1>;
}

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withDictionary('title', PageHeader);
