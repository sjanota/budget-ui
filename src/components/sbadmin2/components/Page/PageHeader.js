import PropTypes from 'prop-types';
import React from 'react';

import { withDictionary } from '../../language';

function PageHeader({ children, title }) {
  return (
    <h1 className='h3 mb-4 text-gray-800 d-flex justify-content-between'>
      {title}
      {children && <div>{children}</div>}
    </h1>
  );
}

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withDictionary('title', PageHeader);
