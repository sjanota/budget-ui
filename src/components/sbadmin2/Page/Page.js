import React from 'react';
import PropTypes from 'prop-types';
import PageHeader from './PageHeader';

export function Page({ children }) {
  return <div className="container-fluid">{children}</div>;
}

Page.propTypes = {
  children: PropTypes.node,
};

Page.Header = PageHeader;
