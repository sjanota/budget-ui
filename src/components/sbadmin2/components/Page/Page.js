import PropTypes from 'prop-types';
import React from 'react';

import PageHeader from './PageHeader';

export default function Page({ children }) {
  return <div className='container-fluid d-flex flex-column'>{children}</div>;
}

Page.propTypes = {
  children: PropTypes.node,
};

Page.Header = PageHeader;
