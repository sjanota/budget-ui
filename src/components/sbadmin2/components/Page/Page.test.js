import { render } from '@testing-library/react';
import React from 'react';

import Page from './Page';

describe('Page', () => {
  it('renders its children', () => {
    const { queryByText } = render(
      <Page>
        <Page.Header title='My page' />
        <p>content</p>
        <p>and more content</p>
      </Page>
    );
    expect(queryByText('My page')).toBeTruthy();
    expect(queryByText('content')).toBeTruthy();
    expect(queryByText('and more content')).toBeTruthy();
  });
});
