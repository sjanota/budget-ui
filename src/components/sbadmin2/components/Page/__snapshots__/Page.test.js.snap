// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders with minimal props 1`] = `
<div
  className="container-fluid"
>
  <h1
    className="h3 mb-4 text-gray-800"
  >
    My page
  </h1>
  <p>
    content
  </p>
  <p>
    and more content
  </p>
</div>
`;
